server/app/services/elasticsearch_service.py

```diff
@@ -0,0 +1,225 @@
+            logger.error(f"Error creating index: {str(e)}")
+
+    @staticmethod
+    def delete_index(es: Elasticsearch, index_name: str):
+        """Delete an Elasticsearch index if it exists."""
+        try:
+            if es.indices.exists(index=index_name):
+                logger.info(f"Deleting existing index: {index_name}")
+                es.indices.delete(index=index_name)
+                logger.info(f"Successfully deleted index: {index_name}")
+            else:
+                logger.info(f"Index {index_name} does not exist, no need to delete")
+        except Exception as e:
+            logger.error(f"Error deleting index: {str(e)}")
+            raise
+
+    @staticmethod
+    def clean_content(content: str) -> str:
+        """Clean content by normalizing whitespace and removing excessive newlines."""
+        if not content:
+            return ""
+        # Replace multiple newlines with a single newline
+        content = '\n'.join(line.strip() for line in content.split('\n') if line.strip())
+        # Replace multiple spaces with a single space
+        content = ' '.join(content.split())
+        return content
+
+    @staticmethod
+    def index_document(es: Elasticsearch, content, embedding):
+        try:
+            index_name = os.getenv("ELASTICSEARCH_INDEX", "documents")
+            logger.info(f"Using index: {index_name}")
+            
+            # Validate inputs
+            if not content or not isinstance(content, str):
+                raise ValueError(f"Invalid content type: {type(content)}")
+            if not embedding or not isinstance(embedding, list):
+                raise ValueError(f"Invalid embedding type: {type(embedding)}")
+            if len(embedding) != 768:  # Match nomic-embed-text dimensions
+                raise ValueError(f"Invalid embedding dimensions: {len(embedding)}")
+            
+            # Clean content
+            cleaned_content = ElasticsearchService.clean_content(content)
+            if not cleaned_content:
+                raise ValueError("Content is empty after cleaning")
+            
+            # Check if index exists with wrong mapping
+            try:
+                if es.indices.exists(index=index_name):
+                    mapping = es.indices.get_mapping(index=index_name)
+                    current_dims = mapping[index_name]['mappings']['properties']['embedding'].get('dims')
+                    if current_dims != 768:
+                        logger.info(f"Index exists with wrong dimensions ({current_dims}), recreating...")
+                        ElasticsearchService.delete_index(es, index_name)
+                        ElasticsearchService.create_index(es, index_name)
+            except Exception as e:
+                logger.error(f"Error checking index mapping: {str(e)}")
+                # If there's any error with the existing index, recreate it
+                ElasticsearchService.delete_index(es, index_name)
```

Comment:
For delete index, this could cause issues in production. A different approach is needed to handle this.